// @beyond-js/backend/listen/listen.cjs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.listen = exports.hmr = exports.__beyond_pkg = void 0;
var dependency_0 = require("@beyond-js/kernel/bundle");
var dependency_1 = require("@beyond-js/backend/server");
var dependency_2 = require("http");
const {
  Bundle: __Bundle
} = dependency_0;
const __pkg = new __Bundle({
  "module": {
    "vspecifier": "@beyond-js/backend@0.1.7/listen"
  },
  "type": "ts"
}).package();
;
__pkg.dependencies.update([['@beyond-js/backend/server', dependency_1], ['http', dependency_2]]);
const ims = new Map();

/************************
INTERNAL MODULE: ./listen
************************/

ims.set('./listen', {
  hash: 2537689596,
  creator: function (require, exports) {
    "use strict";

    Object.defineProperty(exports, "__esModule", {
      value: true
    });
    exports.listen = listen;
    var _server = require("@beyond-js/backend/server");
    /*bundle*/
    function listen(port) {
      if (typeof globalThis.__bee === 'object') {
        const specs = globalThis.__bee.specs;
        port = specs.ports.http;
      }
      if (!port) {
        console.log('Port or Server must be specified');
        return;
      }
      const specs = {};
      typeof port === 'number' ? specs.port = port : specs.server = port;
      new _server.BackendServer(specs);
    }
  }
});
__pkg.exports.descriptor = [{
  "im": "./listen",
  "from": "listen",
  "name": "listen"
}];
let listen;

// Module exports
exports.listen = listen;
__pkg.exports.process = function ({
  require,
  prop,
  value
}) {
  (require || prop === 'listen') && (exports.listen = listen = require ? require('./listen').listen : value);
};
const __beyond_pkg = __pkg;
exports.__beyond_pkg = __beyond_pkg;
const hmr = new function () {
  this.on = (event, listener) => void 0;
  this.off = (event, listener) => void 0;
}();
exports.hmr = hmr;
__pkg.initialise(ims);
